src/main/java/org/c4sg/dao/OrganizationDAO.java

```diff
@@ -41,7 +41,22 @@
             " WHERE ((:keyword is null OR LOWER(o.name) LIKE LOWER(CONCAT('%', :keyword, '%'))" +
                 " OR LOWER(o.description) LIKE LOWER(CONCAT('%', :keyword, '%')) OR LOWER(o.country) LIKE LOWER(CONCAT('%', :keyword, '%')))"
                 + " AND (:status is null OR o.status = :status)"
-                + " AND (:categories is null OR o.category in (:categories))"                
+                + " AND (o.category in (:categories))"                
+                + " AND (:open is null )"
+                + ")  ORDER BY o.name ASC";
+    
+    // Issue #1824 - Introduced query instead of performing (:categories) null check on FIND_BY_CRITERIA
+    String FIND_BY_CRITERIA_NO_CATEGORY_FILTER = "SELECT DISTINCT o FROM Project p RIGHT OUTER JOIN p.organization o" +
+            " WHERE ((:keyword is null OR LOWER(o.name) LIKE LOWER(CONCAT('%', :keyword, '%'))" +
+                " OR LOWER(o.description) LIKE LOWER(CONCAT('%', :keyword, '%')) OR LOWER(o.country) LIKE LOWER(CONCAT('%', :keyword, '%')))"
+                + " AND (:status is null OR o.status = :status)"             
+                + " AND (:open is null )"
+                + ")  ORDER BY o.name ASC";
+    
+    String FIND_BY_CRITERIA_NO_CATEGORIES = "SELECT DISTINCT o FROM Project p RIGHT OUTER JOIN p.organization o" +
```

Comment:
@fumba this FIND_BY_CRITERIA_NO_CATEGORIES is not being called !!! Could you check it ?

Reply:
@felipealves-gnu  Thanks for catching that!!! FIND_BY_CRITERIA_NO_CATEGORIES is the same query asFIND_BY_CRITERIA_NO_CATEGORY_FILTER. I was using FIND_BY_CRITERIA_NO_CATEGORIES at some point as I was debugging the issue. I meant to remove it. 
I have committed a new version of the java class with this query removed.

Reply:
OK, I will add some modifications to your branch as well